app/controllers/shipit/api/release_statuses_controller.rb

```diff
@@ -0,0 +1,23 @@
+module Shipit
+  module Api
+    class ReleaseStatusesController < BaseController
+      require_permission :deploy, :stack
+
+      params do
+        requires :status, String
+      end
+      def create
+        deploy = stack.deploys_and_rollbacks.find(params[:deploy_id])
+        case params[:status]
+        when 'success'
+          deploy.report_healthy!(user: current_user)
+        when 'failure'
+          deploy.report_faulty!(user: current_user)
+        else
+          param_error!(:status, 'Invalid status')
```

Comment:
Instead of raising here, you can do:

```ruby
params do
  requires :status, String
  validates :status, inclusion: { in: %w(success failure) }
end
```

This way explicit-parameters will handle validation for you.

Reply:
Ah cool!